rsocket-core/src/main/java/io/rsocket/plugins/RequestInterceptor.java

```diff
@@ -76,4 +78,27 @@
    * @param metadata taken from the initial frame
    */
   void onReject(Throwable rejectionReason, FrameType requestType, @Nullable ByteBuf metadata);
+
+  @Nullable
+  static RequestInterceptor compose(RequestInterceptor... interceptors) {
+    switch (interceptors.length) {
+      case 0:
+        return null;
+      case 1:
+        return new CompositeRequestInterceptor.SafeRequestInterceptor(interceptors[0]);
+      default:
+        return new CompositeRequestInterceptor(
+            Arrays.stream(interceptors)
+                .flatMap(
+                    ri ->
+                        ri instanceof CompositeRequestInterceptor
+                            ? Arrays.stream(((CompositeRequestInterceptor) ri).requestInterceptors)
+                            : ri instanceof CompositeRequestInterceptor.SafeRequestInterceptor
+                                ? Stream.of(
+                                    ((CompositeRequestInterceptor.SafeRequestInterceptor) ri)
+                                        .requestInterceptor)
+                                : Stream.of(ri))
+                .toArray(RequestInterceptor[]::new));
+    }
+  }
```

Comment:
This could move to `InitializingInterceptorRegistry` which is already advertised as "not for direct use" and is used by `RSocketConnector` and `RSocketServer`.